csrc/device_lower/pass/allocation.cpp

```diff
@@ -670,8 +670,10 @@ class AllocationInserter : public kir::ExprMutator {
       kir::Allocate* mbarrier_alloc =
           IrBuilder::create<kir::Allocate>(mbarrier, MemoryType::Shared);
 
-      auto mbarrier_init = initializeMbarrier(fl, mbarrier);
-      auto mbarrier_inval = invalidateMbarrier(fl, mbarrier);
+      // Initialize and invalidate mbarriers that are used to notify that
+      // the load of the circular buffer is complete.
+      auto mbarrier_init_filled = initializeMbarrier(fl, mbarrier);
+      auto mbarrier_inval_filled = invalidateMbarrier(fl, mbarrier);
```

Comment:
For warp specialization, there will be another mbarrier for notifying the completion of the compute (i.e., the buffer is finished using, and can be loaded with new data again).